### ISSUE TITLE
Official FoxESS API Documentation

### ISSUE THREAD

**Issue by trcalgaro** [macxq/foxess-ha#74]
I share the Official API document received by a technician as a contribution.

[FoxESS Cloud Third-party API Documentation-HTTP(s).zh-CN.en (1).pdf](https://github.com/macxq/foxess-ha/files/9560266/FoxESS.Cloud.Third-party.API.Documentation-HTTP.s.zh-CN.en.1.pdf)

**Comment by macxq** [comment]
```
Signature Calculation
signature=IntegratorID.HmacSHA1(IntegratorSecret,{path}${lang}${string(timestamp)})
User's IntegratorID: AAABBB
IntegratorSecret: CCDDEE
User calls:
https://www.maitian-yun.com/o/v0/inverter/list
lang: zh_CN
timestamp: 12313213123
```

Thx @trcalgaro  for that file, it requires to pass `IntegratorSecret` in the request, can you ask technician how to get it?

**Comment by mhzawadi** [comment]
@macxq I have some working curl commands on issue #72 
Im also playing with getting the data into MQTT

**Comment by trcalgaro** [comment]
> ```
> Signature Calculation
> signature=IntegratorID.HmacSHA1(IntegratorSecret,{path}${lang}${string(timestamp)})
> User's IntegratorID: AAABBB
> IntegratorSecret: CCDDEE
> User calls:
> https://www.maitian-yun.com/o/v0/inverter/list
> lang: zh_CN
> timestamp: 12313213123
> ```
> 
> Thx @trcalgaro for that file, it requires to pass `IntegratorSecret` in the request, can you ask technician how to get it?

I'm waiting for a response from the technician

According to him, he is checking directly with FoxEss CN

**Comment by Paranoidre** [comment]
> > ```
> > Signature Calculation
> > signature=IntegratorID.HmacSHA1(IntegratorSecret,{path}${lang}${string(timestamp)})
> > User's IntegratorID: AAABBB
> > IntegratorSecret: CCDDEE
> > User calls:
> > https://www.maitian-yun.com/o/v0/inverter/list
> > lang: zh_CN
> > timestamp: 12313213123
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Thx @trcalgaro for that file, it requires to pass `IntegratorSecret` in the request, can you ask technician how to get it?
> 
> I'm waiting for a response from the technician
> 
> According to him, he is checking directly with FoxEss CN

Hello! I'm facing a similar situation. Where can I find the "IntegratorSecret" and "IntegratorID"? I'm not a Dev, but I have a customer that's developing a tool and I'm trying to help him. I tried reaching for the foxess support in my country but they just won't answer me.

**Comment by djdarlo** [comment]
Did you manage to get it into MQTT?

**Comment by mhzawadi** [comment]
I have a script that kind of works, but when the cloud service has a blip my script errors. WIll try and update the github repo with my current script

**Comment by guba91** [comment]
mhzawadi, macxq
i'm trying to get the foxess inverter in HA but with "foxess-ha" there is the issue with the deprecated DEVICE_CLASS
with the docker image i'm getting the error `Error response from daemon: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "/foxess/foxess.sh": stat /foxess/foxess.sh: no such file or directory: unknown`
can you help me? :)) thanks!

**Comment by mhzawadi** [comment]
> mhzawadi, macxq i'm trying to get the foxess inverter in HA but with "foxess-ha" there is the issue with the deprecated DEVICE_CLASS with the docker image i'm getting the error `Error response from daemon: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "/foxess/foxess.sh": stat /foxess/foxess.sh: no such file or directory: unknown` can you help me? :)) thanks!

Hi @guba91 
I have fixed running the image, could you take a look and if you still have issues please open an issue on the [git repo](https://github.com/mhzawadi/foxess-mqtt)?

**Comment by guba91** [comment]
> > mhzawadi, macxq i'm trying to get the foxess inverter in HA but with "foxess-ha" there is the issue with the deprecated DEVICE_CLASS with the docker image i'm getting the error `Error response from daemon: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "/foxess/foxess.sh": stat /foxess/foxess.sh: no such file or directory: unknown` can you help me? :)) thanks!
> 
> Hi @guba91 
> I have fixed running the image, could you take a look and if you still have issues please open an issue on the [git repo](https://github.com/mhzawadi/foxess-mqtt)?

Hi @mhzawadi,
Thanks for the update! Now the docker container work fine!
I'm on mobile for today so my ability to tests things are pretty reduced.. What topic shuould i get the data in mqtt?
Thanks!
Btw i've updated foxess-ha, are you able to try my fork to test it? It forks for me, shuould be working.. 👍

**Comment by mhzawadi** [comment]
Nice work @guba91 , your repo works in HA 2023.1.4. Now to see if they break anything in 2023.1.6

**Comment by macxq** [comment]
fixed in v0.16 thanks to @guba91
